### PR TITLE
`Development`: Improve quiz exercise logging

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
### Motivation and Context
Improve logging to understand better what's happening on the production systems

### Steps for Testing
Only code review is needed!

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java [hunk 1]
```diff
@@ -309,9 +311,9 @@ public void scheduleQuizStart(final long quizExerciseId) {
                     // this is expected if we run on multiple nodes
                 }
             }
+            // Do that at the end because this runs asynchronously and could interfere with the cache write above
+            updateQuizExercise(quizExercise);
         }
-        // Do that at the end because this runs asynchronously and could interfere with the cache write above
-        updateQuizExercise(quizExercise);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java:314]: Why was this change needed?

**Review by Strohgelaender — COMMENTED** [review]
Code looks good, only one change where I need a bit more context:

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java (hunk 1 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java:314]: good point: the idea was to prevent a potential null pointer exception in case quizExercise is null, but I overlooked the second check in the if statement a couple lines before

**Review by Strohgelaender — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good

**Review by bassner — APPROVED** [review]
Code LGTM
